source/glest_game/type_instances/unit.cpp

```diff
@@ -2039,6 +2039,12 @@ void Unit::born(const CommandType *ct) {
 	faction->applyStaticProduction(type,ct);
 	setCurrSkill(scStop);
 
+	if(ct != NULL && (ct->getClass() == ccBuild || ct->getClass() == ccRepair)) {
+		//assume this is a building. This could lead to problems in the future,
+		//but there is no real way to detect a building
```

Comment:
That's maybe not a bad thing if it detects everything, not just buildings. If players also get notified when a Thor statue is completed, or a wicker behemoth, that could be helpful too.

But then the output message may need to be more generic.